irohad/synchronizer/impl/synchronizer_impl.cpp

```diff
@@ -95,29 +95,49 @@ namespace iroha {
 
       // TODO andrei 17.10.18 IR-1763 Add delay strategy for loading blocks
       for (const auto &public_key : public_keys) {
-        log_->debug(
-            "trying to download blocks from {} to {} from peer with key {}",
-            my_height + 1,
-            target_height,
-            public_key);
-        auto network_chain =
-            block_loader_
-                ->retrieveBlocks(my_height, PublicKeyHexStringView{public_key})
-                .tap([&my_height](
-                         const std::shared_ptr<shared_model::interface::Block>
-                             &block) { my_height = block->height(); });
-
-        if (validator_->validateAndApply(network_chain, *storage)) {
-          if (my_height >= target_height) {
-            return mutable_factory_->commit(std::move(storage));
+        while (true) {
```

Comment:
Is it possible to merge public keys iterations and tries in a single loop? `for` can be replaced by some conditional increment inside a single loop

Reply:
the variant I implemented seemed simpler to me.